eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs

```diff
@@ -376,64 +412,56 @@ hashAlonzoTxAuxData ::
 hashAlonzoTxAuxData x = AuxiliaryDataHash (hashAnnotated x)
 
 validateAlonzoTxAuxData ::
-  (Era era, ToCBOR (Script era), Script era ~ AlonzoScript era) =>
+  Era era =>
   ProtVer ->
   AuxiliaryData era ->
   Bool
-validateAlonzoTxAuxData pv (AlonzoTxAuxData metadata scrips) =
-  all validMetadatum metadata
-    && all (validScript pv) scrips
+validateAlonzoTxAuxData pv auxData@AlonzoTxAuxData {atadMetadata} =
+  all validMetadatum atadMetadata
+    && all (validScript pv) (getAlonzoTxAuxDataScripts auxData)
 
 instance (EraCrypto era ~ c) => HashAnnotated (AuxiliaryData era) EraIndependentTxAuxData c where
-  hashAnnotated (AuxiliaryDataConstr mb) = mbHash mb
+  hashAnnotated = getMemoSafeHash
 
-deriving newtype instance NFData (Script era) => NFData (AuxiliaryData era)
+deriving newtype instance NFData (AuxiliaryData era)
 
 deriving instance Eq (AuxiliaryData era)
 
-deriving instance (Show (Script era), HashAlgorithm (HASH (EraCrypto era))) => Show (AuxiliaryData era)
+deriving instance (HashAlgorithm (HASH (EraCrypto era))) => Show (AuxiliaryData era)
 
-type instance MemoHashIndex AuxiliaryDataRaw = EraIndependentTxAuxData
+type instance MemoHashIndex AlonzoTxAuxDataRaw = EraIndependentTxAuxData
 
-deriving via InspectHeapNamed "AuxiliaryDataRaw" (AuxiliaryData era) instance NoThunks (AuxiliaryData era)
+deriving via
+  InspectHeapNamed "AlonzoTxAuxDataRaw" (AuxiliaryData era)
+  instance
+    NoThunks (AuxiliaryData era)
 
 deriving via
-  (Mem AuxiliaryDataRaw era)
+  (Mem AlonzoTxAuxDataRaw era)
   instance
-    ( Era era,
-      FromCBOR (Annotator (Script era)),
-      AlonzoScript era ~ Script era -- FIXME: this smells fishy
```

Comment:
No more fishy smell

Reply:
maybe it was from Shelley :shell: .

:drum: